Cargo.toml

```diff
@@ -15,6 +15,8 @@ edition = "2018"
 travis-ci = { repository = "i2p/i2p-rs" }
 
 [dependencies]
+failure = "0.1"
+failure_derive = "0.1"
```

Comment:
I'm not a fan of using `failure` in library crates, but I'll let this by for now (we can always remove it later).